src/core/MOM_porous_barriers.F90

```diff
@@ -125,32 +121,39 @@ subroutine por_widths(h, tv, G, GV, US, eta, pbv, eta_bt, halo_size, eta_to_m)
 end subroutine por_widths
 
 !> subroutine to calculate the profile fit for a single layer in a column
-subroutine calc_por_layer(D_min, D_max, D_avg, eta_layer, w_layer, A_layer)
+subroutine calc_por_layer(D_min, D_max, D_avg, eta_layer, w_layer, A_layer, Z_ref, Z_to_m)
 
-  real,            intent(in) :: D_min !< minimum topographic height [m]
-  real,            intent(in) :: D_max !< maximum topographic height [m]
-  real,            intent(in) :: D_avg !< mean topographic height [m]
-  real,            intent(in) :: eta_layer !< height of interface [m]
+  real,            intent(in)  :: D_min !< minimum topographic height [Z ~> m]
+  real,            intent(in)  :: D_max !< maximum topographic height [Z ~> m]
+  real,            intent(in)  :: D_avg !< mean topographic height [Z ~> m]
+  real,            intent(in)  :: eta_layer !< height of interface [Z ~> m]
   real,            intent(out) :: w_layer !< frac. open interface width of current layer [nondim]
-  real,            intent(out) :: A_layer !< frac. open face area of current layer [nondim]
+  real,            intent(out) :: A_layer !< frac. open face area of current layer [Z ~> m]
+  real,            intent(in)  :: Z_ref !< reference value for geometric height fields [Z ~> m]
+  real,            intent(in)  :: Z_to_m !< a unit conversion factor from units of Z to m
   !local variables
-  real m, a, &   !convenience constant for fit [nondim]
-       zeta, &   !normalized vertical coordinate [nondim]
-       psi, &    !fractional width of layer between Dmin and Dmax [nondim]
-       psi_int !integral of psi from 0 to zeta
+  real Dmin, Dmax, Davg, & !copies of input topographic heights stored in [m]
+       etam, &             !copy of eta_layer stored in [m]
+       m, a, &             !convenience constant for fit [nondim]
+       zeta, &             !normalized vertical coordinate [nondim]
+       psi, &              !fractional width of layer between Dmin and Dmax [nondim]
+       psi_int             !integral of psi from 0 to zeta
+
```

Comment:
You shouldn't have to rescale any of the variables in this routine back to [m].  All of them come in with units of Z, and all should be returned with [nondim] units for w_layer, and [Z ~> m] for A_layer.  So all of the intermediate variables could also have units of [Z ~> m], and you would not need any rescaling factors at all in this routine.

  In fact, the lack of rescaling factors is is one of the ways that you know your code is correct.  The other is to test your code with various values of Z_RESCALE_POWER, T_RESCALE_POWER, etc., and verify that you get identical answers.  (I suspect that you have done the runs for the rescaling tests, because the code in por_widths() is written as though A_layer has units of [m], not [Z ~> m] as documented in the comments.)

Reply:
Thank you for the comments @Hallberg-NOAA. Yes, I ran the *_RESCALE_POWER tests manually to check. I see what you mean though and I agree it's more elegant without the rescaling. I'll update that!